### PR TITLE
default tgo spmd

### PR DESCRIPTION
**Pull request by tommydcjung**
- use default tile-group origin.
- remove defunct targets, options.

### CODE REVIEW

File: software/spmd/bsg_print_stat/Makefile [hunk 1]
```diff
@@ -1,5 +1,3 @@
-bsg_tiles_org_X =0
-bsg_tiles_org_Y =1
```
Comment (drichmond) [inline software/spmd/bsg_print_stat/Makefile]: Why are these no longer necessary? Do we set them in the SPMD loader now? (sort of like how we set them in CUDA)

File: software/spmd/instr_tests/branch_rv32/Makefile [hunk 2]
```diff
@@ -1,6 +1,3 @@
-bsg_tiles_X= 4 
-bsg_tiles_Y= 4 
-
```
Comment (drichmond) [inline software/spmd/instr_tests/branch_rv32/Makefile]: In other places, you added bsg_tiles_X = 1 and bsg_tiles_Y = 1. Why did you add them in some places and remove them in others?

File: software/spmd/nprimes/Makefile [hunk 3]
```diff
@@ -23,23 +7,11 @@ bsg_tiles_Y = 1
...
 
 OBJECT_FILES=main.o
 
 include ../Makefile.include
 
-#########################################################
-#            FPU OPTION
-#     The number of horizon node must be two and must 
-#     be vanilla core 
-BSG_FPU_OP=0
```
Comment (drichmond) [inline software/spmd/nprimes/Makefile]: Out of curiosity, what was BSG_FPU_OP originally for? When we had two versions of Manycore?

File: software/spmd/vcache_atomic_inc/Makefile [hunk 4]
```diff
@@ -1,5 +1,3 @@
-bsg_tiles_org_X =0
-bsg_tiles_org_Y =1
 #bsg_tiles_X = 1
 #bsg_tiles_Y = 1
```
Comment (drichmond) [inline software/spmd/vcache_atomic_inc/Makefile]: If we're getting rid of unused comments, why not get rid of these too?

**Comment by drichmond** [comment]
Not much controversial here, i'd like to approve it. But, I don't understand why some of the changes are necessary

**Comment by drichmond** [comment]
I think the questions in my comments will help my understanding. Simple questions, I believe

File: software/spmd/bsg_print_stat/Makefile (hunk 1 above)
Comment (tommydcjung) [reply software/spmd/bsg_print_stat/Makefile]: the default is defined in mk/Makefile.dimensions. this is passed to nbf.py to create nbf file.

File: software/spmd/nprimes/Makefile (hunk 3 above)
Comment (tommydcjung) [reply software/spmd/nprimes/Makefile]: I don't know. you can ask Shaolin

Comment (drichmond) [reply software/spmd/nprimes/Makefile]: I'm asking you though. Do you know why it's deprecated and needs to be removed? Or is it just no longer used anywhere and we believe we can safely remove it?

Comment (drichmond) [reply software/spmd/nprimes/Makefile]: I see it in here, which looks like something for design compiler: 
https://github.com/bespoke-silicon-group/bsg_manycore/blob/master/syn/Makefile.dc

Comment (tommydcjung) [reply software/spmd/nprimes/Makefile]: It's not used as anywhere. that Makefile is also not used.

File: software/spmd/instr_tests/branch_rv32/Makefile (hunk 2 above)
Comment (tommydcjung) [reply software/spmd/instr_tests/branch_rv32/Makefile]: I add it back for this. Most tests are meant to run only on a single tile.

File: software/spmd/bsg_print_stat/Makefile (hunk 1 above)
Comment (tommydcjung) [reply software/spmd/bsg_print_stat/Makefile]: This is so that it makes it easier to switch the coordinate for the top-left tile, so we can just change the Makefile.dimension, instead of modifying every Makefile.

File: software/spmd/nprimes/Makefile (hunk 3 above)
Comment (drichmond) [reply software/spmd/nprimes/Makefile]: OK, so BSG_FPU_OP is no longer used. If you're going to remove it let's remove it everywhere. 

```software/spmd/bsg_cache_miss/Makefile:BSG_FPU_OP=0
software/spmd/bsg_cuda_lite_runtime/softmax/Makefile:BSG_FPU_OP=0
software/spmd/bsg_cuda_lite_runtime/log_softmax/Makefile:BSG_FPU_OP=0
software/spmd/bsg_cuda_lite_runtime/conv1d/Makefile:BSG_FPU_OP=0
software/spmd/bsg_hetero_nodes/Makefile:BSG_FPU_OP=0
software/spmd/bypass_core/Makefile:#BSG_FPU_OP=1
software/spmd/crc32/Makefile:BSG_FPU_OP=0
software/spmd/depend_stall_mispredict/Makefile:BSG_FPU_OP=0
software/spmd/deprecated/float1/Makefile:BSG_FPU_OP=1
software/spmd/deprecated/float2/Makefile:BSG_FPU_OP=1
software/spmd/deprecated/shared_fam/Makefile:BSG_FPU_OP=1
software/spmd/dram_addr_mapping/Makefile:BSG_FPU_OP=0
software/spmd/fft/Makefile:BSG_FPU_OP=0
software/spmd/fhello_malloc/Makefile:BSG_FPU_OP=0
software/spmd/fib/Makefile:BSG_FPU_OP=0
software/spmd/float_math/Makefile:BSG_FPU_OP=0
software/spmd/mul_div/Makefile:#BSG_FPU_OP=1
software/spmd/nprimes/Makefile:BSG_FPU_OP=0
software/spmd/quicksort/Makefile:BSG_FPU_OP=0
software/spmd/remote_load_crc/Makefile:BSG_FPU_OP=0
software/spmd/specint2000/Makefile.common:BSG_FPU_OP = 0
software/spmd/striped_hello/Makefile:BSG_FPU_OP=0
software/spmd/striped_struct_vector/Makefile:BSG_FPU_OP=0
software/spmd/striped_vector_sum/Makefile:BSG_FPU_OP=0```

File: software/spmd/bsg_print_stat/Makefile (hunk 1 above)
Comment (drichmond) [reply software/spmd/bsg_print_stat/Makefile]: Makes sense, thanks for explaining!

File: software/spmd/instr_tests/branch_rv32/Makefile (hunk 2 above)
Comment (drichmond) [reply software/spmd/instr_tests/branch_rv32/Makefile]: Is there a default bsg_tiles_X/Y?

File: software/spmd/crc32/Makefile [hunk 5]
```diff
@@ -23,23 +21,11 @@ bsg_tiles_Y = 2
 
 all: main.run
 
-# Rule to write processor execution logs. To be used after the
-# verilog simulation.
-#
-# Redirects verilog standard output starting with "X<x_cord>_Y<y_cord>.pelog" 
-# to a unique log file for each coordinate in the manycore. This can be useful 
-# for a quick debug of processor or program running on it.
-proc_exe_logs: X0_Y0.pelog X1_Y0.pelog
```
Comment (drichmond) [inline software/spmd/crc32/Makefile]: Ditto for proc_exe_logs: 
```spmd/bsg_cuda_lite_runtime/log_softmax/Makefile:proc_exe_logs: X0_Y0.pelog X1_Y0.pelog
spmd/bsg_cuda_lite_runtime/conv1d/Makefile:proc_exe_logs: X0_Y0.pelog X1_Y0.pelog
spmd/bsg_hetero_nodes/Makefile:proc_exe_logs: X0_Y0.pelog X1_Y0.pelog
spmd/bsg_remote_congestion/Makefile:proc_exe_logs: X0_Y0.pelog X0_Y5.pelog
spmd/bsg_riscv_tests/Makefile.include:proc_exe_logs: X0_Y0.pelog X0_Y1.pelog X1_Y0.pelog X1_Y1.pelog
spmd/crc32/Makefile:proc_exe_logs: X0_Y0.pelog X1_Y0.pelog
spmd/depend_stall_mispredict/Makefile:proc_exe_logs: X0_Y0.pelog X1_Y0.pelog
spmd/fhello_malloc/Makefile:proc_exe_logs: X0_Y2.pelog
spmd/fib/Makefile:proc_exe_logs: X0_Y0.pelog X1_Y0.pelog
spmd/float_math/Makefile:proc_exe_logs: X0_Y0.pelog X1_Y0.pelog
spmd/load_dependency/Makefile:proc_exe_logs: X0_Y0.pelog X1_Y0.pelog X0_Y1.pelog X1_Y1.pelog
spmd/nprimes/Makefile:proc_exe_logs: X0_Y0.pelog X1_Y0.pelog
spmd/quicksort/Makefile:proc_exe_logs: X0_Y0.pelog X1_Y0.pelog
spmd/regfile_x0_test/Makefile:proc_exe_logs: X0_Y0.pelog X1_Y0.pelog X0_Y1.pelog X1_Y1.pelog
spmd/remote_load_crc/Makefile:proc_exe_logs: X0_Y0.pelog
```

File: software/spmd/nprimes/Makefile [hunk 6]
```diff
@@ -1,19 +1,3 @@
...
-#Tile group configuration
-# If not configured, Will use default Values
-#	bsg_tiles_org_X ?= 0
-#	bsg_tiles_org_Y ?= 1
-bsg_tiles_org_X =1
-bsg_tiles_org_Y =1
-
 # If not configured, Will use default Values
 #	bsg_tiles_X ?= 2
 #	bsg_tiles_Y ?= 2
```
Comment (drichmond) [inline software/spmd/nprimes/Makefile]: is nprimes supposed to run on a 1x1 tile group, or a 2x2 tile group?

**Comment by drichmond** [comment]
A general question: Should we always specify Tile Group dimensions? If not, when should we? Should we leave those lines commented? 

This seems to be ad-hoc at the moment and I think it's worth a thought

File: software/spmd/instr_tests/branch_rv32/Makefile (hunk 2 above)
Comment (tommydcjung) [reply software/spmd/instr_tests/branch_rv32/Makefile]: the default is the full array.

File: software/spmd/nprimes/Makefile (hunk 6 above)
Comment (tommydcjung) [reply software/spmd/nprimes/Makefile]: 1x1

Comment (drichmond) [reply software/spmd/nprimes/Makefile]: But the default is 4x4 and tiles_X tiles_Y isn't specified?

**Comment by tommydcjung** [comment]
In spmd, the default is the full array. The test should be written so that it works for any size of array. If not, (i.e. it runs only on a single tile), then specify the dim in makefile.

**Comment by drichmond** [comment]
> In spmd, the default is the full array. The test should be written so that it works for any size of array. If not, (i.e. it runs only on a single tile), then specify the dim in makefile.

Is this written anywhere?

File: software/spmd/bypass_core/Makefile [hunk 7]
```diff
@@ -2,17 +2,6 @@
 bsg_tiles_X = 4
 bsg_tiles_Y = 4
 
-#########################################################
-#each bytes defines the core type
-# 00: vanilla core
-# 01  vscale  core
-TYPE_VEC="64'h00_00_00_00"
```
Comment (drichmond) [inline software/spmd/bypass_core/Makefile]: Is type_vec still a thing we support?

Comment (tommydcjung) [reply software/spmd/bypass_core/Makefile]: nope

Comment (tommydcjung) [reply software/spmd/bypass_core/Makefile]: This is related to the machine config. It should be supported in machine.include.

Comment (drichmond) [reply software/spmd/bypass_core/Makefile]: OK, so we have that in other makefiles too, right?

**Comment by drichmond** [comment]
Looks good! My comment about documentation of default manycore dimensions still stands, but this PR is good

**Comment by drichmond** [comment]
My comments are more relevant to https://github.com/bespoke-silicon-group/bsg_manycore/issues/222

**Comment by taylor-bsg** [comment]
Quick question — about type vector; isn’t Cornell using this for their
accelerators?

M

**Comment by tommydcjung** [comment]
Nope it's not passed into vcs. It should be something configured in Makefile.machine.include
